var/spack/repos/builtin/packages/intel-parallelstudio/package.py

```diff
@@ -0,0 +1,112 @@
+    variant('mkl', default=True, description="Install the Intel MKL library")
+    variant('daal', default=True, description="Install the Intel DAAL libraries")
+    variant('ipp', default=True, description="Install the Intel IPP libraries")
+    variant('tools', default=True, description="Install the Intel Advisor, VTune Amplifier, and Inspector tools")
+
+    provides('mpi', when='@cluster:+mpi')
+    provides('mkl', when='+mkl')
+    provides('daal', when='+daal')
+    provides('ipp', when='+ipp')
+
+    def install(self, spec, prefix):
+
+        # remove the installation DB, otherwise it will try to install into location of other Intel builds
+        try:
+            os.remove(os.path.join(os.environ["HOME"], "intel", "intel_sdp_products.db"))
+        except OSError:
+            pass # if the file does not exist
+
+        base_components = "ALL" # when in doubt, install everything
+        mpi_components = ""
+        mkl_components = ""
+        daal_components = ""
+        ipp_components = ""
+        tools_components = ""
+
+        if spec.satisfies('+all'):
+            base_components = "ALL"
+        else:
+            with open("pset/mediaconfig.xml", "r") as f:
+                lines = f.readlines()
+                all_components = []
+                for line in lines:
+                    if line.find('<Abbr>') != -1:
+                        component = line[line.find('<Abbr>') + 6:line.find('</Abbr>')]
+                        all_components.append(component)
+                base_components = filter_pick(all_components, re.compile('(comp|openmp|intel-tbb|icc|ifort|psxe|icsxe-pset)').search)
+                mpi_components = filter_pick(all_components, re.compile('(icsxe|imb|mpi|itac|intel-tc|clck)').search)
+                mkl_components = filter_pick(all_components, re.compile('(mkl)').search)
+                daal_components = filter_pick(all_components, re.compile('(daal)').search)
+                ipp_components = filter_pick(all_components, re.compile('(ipp)').search)
+                tool_components = filter_pick(all_components, re.compile('(gdb|vtune|inspector|advisor)').search)
+
+        components = base_components
+        if not spec.satisfies('+all'):
+            if spec.satisfies('+mpi') and 'cluster' in str(spec.version):
+                components += mpi_components
+            if spec.satisfies('+mkl'):
+                components += mkl_components
+            if spec.satisfies('+daal'):
+                components += daal_components
+            if spec.satisfies('+ipp'):
+                components += ipp_components
+            if spec.satisfies('+tools') and (spec.satisfies('@cluster') or spec.satisfies('@professional')):
+                components += tool_components
+
+        self.intel_components = ';'.join(components)
+        IntelInstaller.install(self, spec, prefix)
+
+        absbindir = os.path.split(os.path.realpath(os.path.join(self.prefix.bin, "icc")))[0]
```

Comment:
Oh, so you're saying that bin isn't a symlink, the files in it are. Gotcha. Then using `os.path.dirname` should be fine.